keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,914 @@
+  - [Possibility for the configurable reason for the &quot;SuccessCriteriaMet&quot; condition](#possibility-for-the-configurable-reason-for-the-successcriteriamet-condition)
+    - [Additional Story](#additional-story-1)
+    - [Job API](#job-api-2)
+      - [Set the entire reason](#set-the-entire-reason)
+      - [Set the suffix of the reason](#set-the-suffix-of-the-reason)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [x] (R) KEP approvers have approved the KEP status as `implementable`
+- [x] (R) Design details are appropriately documented
+- [x] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [x] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP extends the Job API to allow setting conditions under which an Indexed Job can be declared as succeeded.
+
+## Motivation
+
+There are cases where a batch workload requires an indexed job that want to care 
+only about leader indexes in determining the success or failure of a Job,
+for example MPI and PyTorch etc. This is currently not possible because the indexed job 
+is marked as Completed only if all indexes succeeded.
+
+Some third-party frameworks have implemented success policy.
+
+- [Kubeflow Training Operator](https://www.kubeflow.org/docs/components/training)
+- [Flux Operator](https://flux-framework.org/flux-operator/)
+- [JobSet](https://github.com/kubernetes-sigs/jobset/)
+
+### Goals
+
+- Allow to mark a job as a `Succeeded` according to a declared policy.
```

Comment:
```suggestion
- Allow to mark a job as succeeded according to a declared policy.
```
nit: Maybe the use of backquote suggests it is a API term, like condition, but the condition is actually called `Complete`

Reply:
Yes, that's right. Consistency is important.

Reply:
Done.